src/metabase/util/date_2.clj

```diff
@@ -106,13 +106,22 @@
                     (.toFormatter builder))
    OffsetTime     (let [builder (doto (DateTimeFormatterBuilder.)
                                   (.append (DateTimeFormatter/ofLocalizedTime FormatStyle/MEDIUM))
-                                  (.appendPattern " (OOOO)"))]
+                                  (.appendLiteral " (")
+                                  (.appendLocalizedOffset TextStyle/FULL)
+                                  (.appendLiteral ")"))]
                     (.toFormatter builder))
    OffsetDateTime (let [builder (doto (DateTimeFormatterBuilder.)
                                   (.appendLocalized FormatStyle/LONG FormatStyle/MEDIUM)
-                                  (.appendPattern " (OOOO)"))]
+                                  (.appendLiteral " (")
+                                  (.appendLocalizedOffset TextStyle/FULL)
+                                  (.appendLiteral ")"))]
                     (.toFormatter builder))
-   ZonedDateTime  (DateTimeFormatter/ofLocalizedDateTime FormatStyle/LONG)})
+   ZonedDateTime  (let [builder (doto (DateTimeFormatterBuilder.)
+                                  (.appendLocalized FormatStyle/LONG FormatStyle/MEDIUM)
+                                  (.appendLiteral " (")
+                                  (.appendZoneText TextStyle/FULL)
+                                  (.appendLiteral ")"))]
+                    (.toFormatter builder))})
```

Comment:
Tweaked this so it returns a nice version of the timezone e.g. `(Pacific Daylight Time)` instead of a short abbreviation e.g. `PDT`